tools.nix

```diff
@@ -232,11 +249,12 @@ rec {
         mkGitHash = { source, ... }@attrs:
           let
             parsed = parseGitSource source;
-            src = builtins.fetchGit {
+            src = builtins.fetchGit ({
               submodules = true;
               inherit (parsed) url rev;
-              ref = attrs.branch or "master";
-            };
+            } // lib.optionalAttrs (parsed ? branch) {
+              ref = parsed.branch;
```

Comment:
Is the branch always set in the git source URL?

Reply:
It isn't, in which case it just takes the default one (remote side `HEAD`). I say this matter for some `main` vs `master` defaults.

Reply:
Yeah, upon a second reading it makes sense.